src/reducers/bitcoinReducer.js

```diff
@@ -120,6 +146,43 @@ const updateUnspentTransactions = (
       balances,
       addresses: [...filteredAddresses, updatedAddress],
       unspentTransactions: filteredTransactions.concat(...unspentTransactions),
+      transactions: existingTxs,
+    },
+  };
+};
+
+const updateBitcoinTransactions = (
+  state: BitcoinReducerState,
+  action: UpdateBTCTransactionsAction,
+): BitcoinReducerState => {
+  const { transactions: updateTransactions, address } = action;
+  const {
+    data: {
+      addresses, transactions, balances, unspentTransactions,
+    },
+  } = state;
+
+  const filteredTransactions: BTCTransaction[] = transactions.filter(
+    ({ address: txAddress }) => address !== txAddress,
```

Comment:
how about `!addressesEqual(address, txAddress)`?
same for other similar places in this file